### PR TITLE
Add DataDog source

### PR DESCRIPTION
**Pull request by willmarks**
## Description

Implementing the DataDog source with `FULL_REFRESH` and `INCREMENTAL` streams for Incidents and Users.

### Sorry about all the file changes
For all the `test_files/spec.json` files that are being removed, this is because they are all duplicates of the `resources/spec.json` and are not needed if we just references the actual spec. (Confirmed that this was something we wanted to do with @cjwooo). Tests now need to point to the `resource/spec.json`

## In Progress
Im still trying to get the acceptance tests to work...

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/agileaccelerator-source/test/index.test.ts [hunk 1]
```diff
@@ -17,6 +17,10 @@ jest.mock('axios');
 
 const mockedAxios = axios as jest.Mocked<typeof axios>;
 
+function readResourceFile(fileName: string): any {
+  return JSON.parse(fs.readFileSync(`resources/${fileName}`, 'utf8'));
+}
```
Comment (willmarks) [inline sources/agileaccelerator-source/test/index.test.ts:22]: We may want to pull these up into a util in the cdk or something...

File: sources/datadog-source/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import {Incidents, Users} from './streams';
+
+/** The main entry point. */
+export function mainCommand(): Command {
+  const logger = new AirbyteLogger();
+  const source = new DataDogSource(logger);
+  return new AirbyteSourceRunner(logger, source).mainCommand();
+}
+
+/** Example source implementation. */
```
Comment (willmarks) [inline sources/datadog-source/src/index.ts:22]: I'll remove these things

File: sources/datadog-source/resources/spec.json [hunk 3]
```diff
@@ -2,27 +2,27 @@
...
         "type": "string",
-        "title": "StatusPage API key",
+        "title": "DataDog Application Key",
         "airbyte_secret": true
       },
-      "org_id": {
-        "type": "string",
-        "title": "StatusPage Organization ID"
+      "pageSize": {
+        "type": "number",
```
Comment (cjwooo) [inline sources/datadog-source/resources/spec.json:24]: Change this type to "integer" -- Airbyte UI doesn't properly handle number types currently.

Comment (cjwooo) [reply sources/datadog-source/resources/spec.json:24]: Also specify your default page size here to inform the user.

File: sources/datadog-source/test_files/incremental_configured_catalog.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        ],
+        "source_defined_cursor": true,
+        "source_defined_primary_key": [
+          [
+            "id"
+          ]
+        ]
+      },
+      "sync_mode": "incremental",
+      "destination_sync_mode": "overwrite"
```
Comment (cjwooo) [inline sources/datadog-source/test_files/incremental_configured_catalog.json:19]: Change the destination sync modes in this file to "append" (shouldn't actually affect anything, but is more accurate representation of what user would do)

**Review by cjwooo — APPROVED** [review]
Bump the package version to 0.1.36.

**Pull request merged**